client/file-manager/src/in_memory.rs

```diff
@@ -154,7 +163,8 @@ where
             .as_str(),
         );
 
-        if metadata.chunks_count() != file_data.stored_chunks_count() {
+        let stored_chunks = file_data.stored_chunks_count()?;
+        if metadata.chunks_count() != stored_chunks {
```

Comment:
Why do we check this? Wouldn't we check initially after adding a file to storage initially when received from a user so we don't have to check everytime aftewards?

Reply:
I guess given the asynchronous nature of this processes, the extra check doesn't hurt.

Reply:
It does because if you have large files, if you have to iterate through the whole thing every time you need to generate a proof for it.

Reply:
Fair point. We can take it into consideration as an optimisation, at the risk of inconsistencies.